Marlin/src/module/tool_change.cpp

```diff
@@ -1591,6 +1593,10 @@ void tool_change(const uint8_t new_tool, bool no_move/*=false*/) {
     // Perform the tool change
     tool_change(migration_extruder);
 
+    #if MIGRATION_SETTINGS
+      RESTORE(tmp_mig_settings);
```

Comment:
No explicit `RESTORE` is needed unless `toolchange_settings` is used again below this line. `REMEMBER` creates an object that automatically does `RESTORE` when it goes out of scope. If that isn't working for some reason then it's a compiler bug.

Reply:
Ok !
I immediatly go to macro declaration to see how you have made this magic trick
Thks